packages/framer-motion/src/value/use-will-change/__tests__/index.test.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from "react"
+import { render } from "../../../../jest.setup"
+import { useWillChange } from ".."
+import { motion, useMotionValue } from "../../.."
+
+describe("useWillChange", () => {
+    test("Applies 'will-change: auto' by default", async () => {
+        const Component = () => {
+            const willChange = useWillChange()
+            return <motion.div style={{ willChange }} />
+        }
+
+        const { container } = render(<Component />)
+        expect(container.firstChild).toHaveStyle("will-change: auto;")
+    })
+
+    test("Adds externally-provided motion values", async () => {
+        const Component = () => {
+            const willChange = useWillChange()
+            const height = useMotionValue("height")
+            return <motion.div style={{ willChange, height }} />
+        }
+
+        const { container } = render(<Component />)
+
+        return new Promise<void>((resolve) => {
+            requestAnimationFrame(() => {
+                requestAnimationFrame(() => {
+                    expect(container.firstChild).toHaveStyle(
+                        "will-change: height;"
+                    )
+                    resolve()
+                })
+            })
+        })
+    })
+
+    test("Adds 'transform' when transform is animating", async () => {
+        const Component = () => {
+            const willChange = useWillChange()
+            return (
+                <motion.div
+                    animate={{ x: 100, backgroundColor: "#000" }}
+                    style={{ willChange }}
+                />
+            )
+        }
+
+        const { container } = render(<Component />)
+
+        return new Promise<void>((resolve) => {
+            requestAnimationFrame(() => {
+                requestAnimationFrame(() => {
+                    expect(container.firstChild).toHaveStyle(
+                        "will-change: transform, background-color;"
+                    )
+                    resolve()
+                })
+            })
+        })
+    })
+
+    test("Removes `transform` when all transforms finish animating", async () => {
```

Comment:
We could maybe cover one more test to verify that when all the motion values (not just transforms) finish animating, will change is revert to "auto"?